### PR TITLE
Lower amount of network traffic produced in multiplayer & Fix some issues

### PR DESCRIPTION
**Pull request by TSRBerry**
As the title says, this PR aims to lower the amount of network traffic produced in multiplayer.
The following changes were made to achieve that:
- Bots don't sync their inventory anymore, instead it will only be written to modData when saving.
- To sync the hat of each bot we now send a BotHatChangeMessage when the slot changes.

This already makes SV send a lot less data when syncing the bots.
If this still causes problems, I'll change the way we sync positions too, since the change events I added already show that it sends a lot of data while the bots are moving.

### CODE REVIEW

**Comment by TSRBerry** [comment]
Pathoschild told me this when I asked about how to sync the inventory without sending the full xml string to all clients every time:
> You could place a chest off-screen and put the items in it, then it'll synchronize those like usual. [...] just reference the `chest.items` list.

If this works without issues it'll solve a lot of problems we are currently seeing! I'll try to do that before we start a new test.

**Comment by TSRBerry** [comment]
Actually network traffic decreased a lot with a5bd74076e92ba83d09e837af1931e7fa4a5608f, so I'd say if further improvements are necessary we could add the solution Pathoschild was talking about.

**Comment by TSRBerry** [comment]
I'd say this ready for a final review now! 
Now there *finally* shouldn't be any more problems (or at least none introduced with the multiplayer changes).

File: Farmtronics/Bot/BotObject.cs [hunk 1]
```diff
@@ -574,11 +573,8 @@ class BotObject : StardewValley.Object {
 				farmer.tryToMoveInDirection(farmer.FacingDirection, false, 0, false);
 				data.Update();
 				if (TileLocation != farmer.getTileLocation()) {
-					// Remove this object from the Objects list at its old position
-					currentLocation.removeObject(TileLocation, false);
-					// Update our tile pos, and add this object to the Objects list at the new position
-					TileLocation = farmer.getTileLocation();
-					currentLocation.setObject(TileLocation, this);
+					// Move this object to it's new position
```
Comment (JoeStrout) [inline Farmtronics/Bot/BotObject.cs:576]: Sorry, but can you please change "it's" to "its" here?

**Review by JoeStrout — CHANGES_REQUESTED** [review]

**Comment by JoeStrout** [comment]
Great stuff, thank you @TSRBerry !

**Pull request merged**